app/scripts/modules/appengine/serverGroup/configure/wizard/basicSettings.controller.ts

```diff
@@ -53,6 +53,9 @@ class AppengineServerGroupBasicSettingsCtrl implements IController {
     }
 
     const artifact = copy(expected.matchArtifact);
+    if (artifact == null) {
+      return '';
+    }
     return Object.keys(artifact)
```

Comment:
I think there is a filter registered for this, something like summarize-expected-artifacts

Reply:
I'll take another look here.  I saw exceptions in this function around fetching keys for null/undefined.  They occurred after deleting an Expected Artifact from a Configuration stage that was in use in a Deploy stage.